### PR TITLE
Iteration3 AL

### PR DESCRIPTION
**Pull request by alex-latham**
- Includes Max's team_id -> team_name morph
- Adds one method to game_stats
  - `winningest_team`
- Adds three methods to game_teams_stats
  - `percent_difference` (helper method for following two)
  - `best_fans`
  - `worst_fans`
- Fixes typos in GameTeams class and corresponding test file
- Adds new truncated files for tests:
  - `game_teams_truncated_with_best_fans.csv` (original truncated file had multiple teams meeting criteria for "best fans"
  - `games_truncated_with_winningest_team.csv` (original truncated file had multiple teams meeting criteria for "winningest"
- In `games_stats_test`, replace `@games_stats.games.first` with `@game`
- In `team_stats`, make initialization file path and object dynamic
- Use `test_helper` in team_stats_test
- Fix TeamStats object being passed into initialization in `team_stats_test`

### CODE REVIEW

File: data/game_teams_truncated_with_best_fans.csv [hunk 1]
```diff
@@ -0,0 +1,13 @@
+game_id,team_id,HoA,result,settled_in,head_coach,goals,shots,tackles,pim,powerPlayOpportunities,powerPlayGoals,faceOffWinPercentage,giveaways,takeaways
```
Comment (Maxwell-Baird) [inline data/game_teams_truncated_with_best_fans.csv:1]: good job on making different data files so we don't have to remake the test to fit the new information

File: lib/game_teams_stats.rb [hunk 2]
```diff
@@ -7,6 +8,75 @@ class GameTeamStats
...
+    away_games.each do |away_game|
+      if away_game.result == "WIN"
+        away_win_ratios[away_game.team_id][0] += 1
+      end
+        away_win_ratios[away_game.team_id][1] += 1
+    end
+
+    away_win_percentages = away_win_ratios.each_with_object(Hash.new) do |(team_id, away_win_ratio), away_win_percentages|
+      away_win_percentages[team_id] = away_win_ratio[0].fdiv(away_win_ratio[1]) * 100
+    end
```
Comment (Maxwell-Baird) [inline lib/game_teams_stats.rb:49]: We can refactor this later on

File: lib/game_teams.rb [hunk 3]
```diff
@@ -3,20 +3,20 @@ class GameTeams
   attr_reader :game_id,
               :team_id,
               :hoa,
-              :results,
+              :result,
               :settled_in,
-              :head_couch,
+              :head_coach,
```
Comment (whitneykidd) [inline lib/game_teams.rb]: Great catch here 💯

File: data/game_teams_truncated_with_best_fans.csv (hunk 1 above)
Comment (tylertomlinson) [reply data/game_teams_truncated_with_best_fans.csv:1]: I second that

File: lib/game_teams.rb [hunk 4]
```diff
@@ -3,20 +3,20 @@ class GameTeams
   attr_reader :game_id,
               :team_id,
               :hoa,
-              :results,
```
Comment (tylertomlinson) [inline lib/game_teams.rb]: good catch

File: lib/game_teams.rb [hunk 5]
```diff
@@ -3,20 +3,20 @@ class GameTeams
...
+              :result,
               :settled_in,
-              :head_couch,
```
Comment (tylertomlinson) [inline lib/game_teams.rb]: good catch

File: lib/game_teams_stats.rb [hunk 6]
```diff
@@ -7,6 +8,75 @@ class GameTeamStats
 
   def initialize(file_path, object)
     @game_teams = csv_data(file_path, object)
+    # not sure about this
+    @team_stats = TeamStats.new("./data/teams.csv", Team)
```
Comment (tylertomlinson) [inline lib/game_teams_stats.rb:11]: Super easy to make methods in here with this IV

File: test/game_stats_test.rb [hunk 7]
```diff
@@ -15,7 +15,7 @@ def test_it_exists
   end
 
   def test_game_objects_are_new_instances_of_game_class
-    assert_instance_of Game, @games_stats.games.first
+    assert_instance_of Game, @game
```
Comment (tylertomlinson) [inline test/game_stats_test.rb:18]: I like the refactor here.

File: test/game_teams_test.rb [hunk 8]
```diff
@@ -7,10 +7,10 @@ def setup
     @game_teams = GameTeams.new({
           game_id: 2012030221,
           team_id: 3,
-          HoA: "away",
-          results: "LOSS",
+          hoa: "away",
+          result: "LOSS",
           settled_in: "OT",
-          head_couch: "John Tortorella",
+          head_coach: "John Tortorella",
```
Comment (tylertomlinson) [inline test/game_teams_test.rb]: EAGLE EYE™

**Comment by tylertomlinson** [comment]
> * Includes Max's team_id -> team_name morph
> * Adds one method to game_stats
>   
>   * `winningest_team`
> * Adds three methods to game_teams_stats
>   
>   * `percent_difference` (helper method for following two)
>   * `best_fans`
>   * `worst_fans`
> * Fixes typos in GameTeams class and corresponding test file
> * Adds new truncated files for tests:
>   
>   * `game_teams_truncated_with_best_fans.csv` (original truncated file had multiple teams meeting criteria for "best fans"
>   * `games_truncated_with_winningest_team.csv` (original truncated file had multiple teams meeting criteria for "winningest"
> * In `games_stats_test`, replace `@games_stats.games.first` with `@game`
> * In `team_stats`, make initialization file path and object dynamic
> * Use `test_helper` in team_stats_test
> * Fix TeamStats object being passed into initialization in `team_stats_test`

This pull request description is Epic! Good work

**Pull request merged**